### PR TITLE
Fix memory leak that was merged in

### PR DESCRIPTION
**Pull request by permcody**
Make sure that you release memory allocated by `MooseArray`.

I also fixed up several other formatting, warnings and minor issues merged in today (11/23).

### CODE REVIEW

File: test/src/kernels/PotentialAdvection.C [hunk 1]
```diff
@@ -26,9 +26,14 @@ PotentialAdvection::PotentialAdvection(const InputParameters & parameters) :
...
     _sgn(getParam<bool>("positive_charge") ? 1 : -1),
+    _default(_fe_problem.getMaxQps(), RealGradient(-1.)),
     _grad_potential(isCoupled("potential") ? coupledGradient("potential") : _default)
 {
-  _default.resize(_fe_problem.getMaxQps(), RealGradient(-1.));
+}
+
+PotentialAdvection::~PotentialAdvection()
+{
+  _default.release();
```
Comment (permcody) [inline test/src/kernels/PotentialAdvection.C]: This is important!

Comment (friedmud) [inline test/src/kernels/PotentialAdvection.C]: Indeed

**Review by friedmud — APPROVED** [review]
If the tests pass this looks good.

**Comment by permcody** [comment]
@andrsd - can you merge this? Thanks

**Pull request merged**